lib/Dancer2/Core/Request.pm

```diff
@@ -653,6 +653,14 @@ sub _shallow_clone {
     return $new_request;
 }
 
+
+sub route {
+    my ($self, $route) = @_;
+    if ($route) { $self->{route_matched} = $route; }
+
+    return $self->{route_matched};
```

Comment:
Isn't any such optimisation in the Moo'd stuff?  I don't see how it applies here - it's just an attribute in a Request object, which will only live as long as the request being processed.

Interested to see what you come up with, though.